File: Kotlin/58.length_of_last_word.kt
fun main() {
    println(lengthOfLastWord("Hello World"))
    println(lengthOfLastWord("   fly me   to   the moon  "))
    println(lengthOfLastWord("luffy is still joyboy"))
}

fun lengthOfLastWord(s: String): Int {
    var count = 0;
    
    for(i in s.length-1 downTo 0) {
        
        if(s[i].compareTo(' ')==0 && count!=0){
            break;
        } else if(s[i].compareTo(' ')!=0) {
            count++;
        }
    }

    return count;
}

File: lib/58.length_of_last_word.dart
void main() {
  print(lengthOfLastWord("Hello World"));
  print(lengthOfLastWord("   fly me   to   the moon  "));
  print(lengthOfLastWord("luffy is still joyboy"));
}

int lengthOfLastWord(String s) {
  var count = 0;

  for (int i = s.length - 1; i >= 0; i--) {
    if (count != 0 && s[i] == ' ') {
      break;
    } else if (s[i] != ' ') {
      count++;
    }
  }

  return count;
}
